src/app/ReadHandler.cpp

```diff
@@ -103,22 +104,30 @@ CHIP_ERROR ReadHandler::OnInitialRequest(System::PacketBufferHandle && aPayload)
 
     if (err != CHIP_NO_ERROR)
     {
+        Status status = Status::InvalidAction;
+        if (err == CHIP_IM_GLOBAL_STATUS(PathsExhausted))
+        {
+            status = Status::PathsExhausted;
```

Comment:
I think this should be something more like this:
```suggestion
        if (err.IsIMStatus())
        {
            status = StatusIB(err).mStatus;
```
so that we preserve all existing IM status values, not just PathsExhausted, and turn things that are not already IM status into INVALID_ACTION.

Reply:
agree.